pages/home_page.py

```diff
@@ -13,34 +14,40 @@ class CrashStatsHomePage(CrashStatsBasePage):
         Page Object for Socorro
         https://crash-stats.allizom.org/
     """
+
+    _graph_loading_locator = (By.CSS_SELECTOR, '#homepage-graph .loading')
     _release_channels_locator = (By.CSS_SELECTOR, '.release_channel')
     _last_release_channel_locator = (By.CSS_SELECTOR, '#release_channels .release_channel:last-child')
 
-    def __init__(self, testsetup, product=None):
-        '''
-            Creates a new instance of the class and gets the page ready for testing
-        '''
-        CrashStatsBasePage.__init__(self, testsetup)
+    def __init__(self, base_url, selenium, product=None):
+        CrashStatsBasePage.__init__(self, base_url, selenium)
 
         if product is None:
             self.selenium.get(self.base_url)
+        self.wait_for_page_to_load()
+
+    def wait_for_page_to_load(self):
+        Wait(self.selenium, self.timeout).until(
+            lambda s: not self.is_element_present(*self._graph_loading_locator))
+        return self
 
     def click_last_product_top_crashers_link(self):
         return self.ReleaseChannels(
-            self.testsetup, self.selenium.find_element(*self._last_release_channel_locator)
+            self.selenium, self.timeout, self.selenium.find_element(*self._last_release_channel_locator)
```

Comment:
I think this should be `self.base_url, self.selenium` rather than `self.selenium, self.timeout` to be consistent with our other page regions. This affects the changes to the remainder of this file too.